lib/poller.js

```diff
@@ -55,6 +55,19 @@ class Poller extends EventEmitter {
 
   generateSerializedData (splits = []) {
     const latestCache = this.cache
+
+    // If generating serialized data for a subset of splits, get updated segments-related data for the subset
+    const serializingSegmentsForSubset = this.serializeSegments && splits.length !== 0
+    const latestCacheForSubset = {}
+    if (serializingSegmentsForSubset) {
+      Object.assign(latestCacheForSubset, JSON.parse(JSON.stringify(latestCache)))
```

Comment:
why do you need to parse and stringify?

Reply:
This line essentially sets `latestCacheForSubset` to a copy of `latestCache`. This ensures we don't update anything in `latestCache`.

Reply:
I still don't understand why you need to stringify and parse. Object.assign copies the properties from latestCache over to latestCacheForSubset. subsequent changes to latestCacheForSubset do not affect latestCache

Reply:
My main concern here was that `Object.assign` doesn't do a deep copy. Thinking about this more, given that we aren't editing the `segments` object itself, and just having `segments` point to a separate object, we don't need the deep copy logic.